Blog: Our journey to all-remote (#711)
sourcegraph/about: blogposts/our-journey-to-all-remote.md

- Pack up all of our stuff (office supplies, company swag, banners for events, furniture, art and wall accessories, spare electronic equipment and cables, food, and everything in between), and decide whether we should try to store it (and if so, where), sell it, or distribute it to team members.
- Update all of our financial and legal accounts, update contracts, and more.

## Tradeoffs

We love all-remote, but it doesn't work for everyone (just like working on an open-office plan doesn't work for everyone). Working remotely gives you extra freedom and flexibility, but also requires extra discipline and personal responsibility. Certain things that happen organically in an office environment (e.g., informal conversations at lunch), need to be explicitly planned (e.g., by scheduling small "watercooler" meetings with no agenda, or scheduling 1-1s between random teammates using a tool like donut.com).

We are committed to supporting all of our teammates to help them be successful. Check out our [tips for working remotely](https://about.sourcegraph.com/company/remote/tips).

> If you have any questions about all-remote teams, join us during our next Developer Meetup and I will be happy to answer them. [Link here]

Author: @aileenrose can you give me a link to put here?

Author: ```suggestion
If you have any questions about all-remote teams, join us during [our Developer Meetup on 2020-04-07](https://info.sourcegraph.com/online-meetup-april-7) and I will be happy to answer them.
```

---
title: "Our journey to all-remote"
author: Nick Snyder

> publishDate: 2020-03-30T10:00-07:00

Author: ```suggestion
publishDate: 2020-03-31T10:00-07:00
```

tags: [
  blog, remote
]
slug: our-journey-to-all-remote

> heroImage: /blog/sf-office-mural.jpg

Author: @ryan-blunden is this the right place to put a hero image or should it also be in a Google Cloud Bucket?

Reviewer: Yes, that's the correct path. And the location of the image should be: `about/website/static/blog`.

Reviewer 2: @nicksnyder We may look at using a Google Cloud Bucket exclusively at some point, but this is fine for now.

published: true
---

> 

Author: ```suggestion

![Sourcegraph San Francisco office mural](/blog/sf-office-mural.jpg)

```

On February 20, 2020 we officially moved out of our amazing San Francisco office and became an [all-remote](https://about.sourcegraph.com/company/remote) company. Many people have not worked at an all-remote company before and I frequently get questions, especially from candidates, about what led to this decision, so this blog post documents our journey to all-remote.

## Remote friendly engineering

> For most of Sourcegraph's history we have had an office in San Francisco, but many of our early engineering hires were distributed across the world so we have had remote-friendly practices ingrained in our culture from the very beginning. As a result, we strongly value things like asynchronous written communication, personal responsibility, and autonomy; nothing else works when your teammates are in different timezones.

Reviewer 2: ```suggestion
For most of Sourcegraph's history, we have had an office in San Francisco, but many of our early engineering hires were distributed across the world so we have had remote-friendly practices ingrained in our culture from the very beginning. As a result, we strongly value things like asynchronous written communication, personal responsibility, and autonomy; nothing else works when your teammates are in different timezones.
```

> For most of Sourcegraph's history, we have had an office in San Francisco, but many of our early engineering hires were distributed across the world so we have had remote-friendly practices ingrained in our culture from the very beginning. As a result, we strongly value things like asynchronous written communication, personal responsibility, and autonomy; nothing else works when your teammates are in different timezones.

Author: ```suggestion
For most of Sourcegraph's history, we have had an office in San Francisco, but many of our early engineering hires were distributed across the world, so we have had remote-friendly practices ingrained in our culture from the very beginning. As a result, we strongly value things like asynchronous written communication, personal responsibility, and autonomy; nothing else works when your teammates are in different timezones.
```

## Distributed days

> Since we were open to hiring engineers from all over the world, our engineering team organically grew to be more distributed over time. Continuing to be a remote-friendly place to work was important for us as we grew the team. In late 2018 we started experimenting with scheduled days where SF teammates were encouraged to work from a location other than the office; we called these "distributed days".

Reviewer 2: ```suggestion
Since we were open to hiring engineers from all over the world, our engineering team organically grew to be more distributed over time. Continuing to be a remote-friendly place to work was important for us as we grew the team. In late 2018, we started experimenting with scheduled days where SF teammates were encouraged to work from a location other than the office; we called these "distributed days".
```

This experiment had two goals:

> 1. Give SF teammates dedicated heads downtime to optimize productivity (e.g., fewer interruptions, skip your commute).

Reviewer 2: > Give SF teammates dedicated heads downtime 

This reads awkwardly for me, but perhaps this is a saying I'm not familiar with.

Author: Does this sound better to you?

```suggestion
1. Give SF teammates more focus time to optimize productivity (e.g., fewer interruptions, skip your commute).
```

    - Distributed best practices should be documented and practiced as part of our culture.

## Remote-first

In the middle of 2019, we decided to more formally communicate to the team that we wanted to be remote-first.

In practice, this meant:

- There was no expectation that teammates within commuting distance of our SF office needed to show up at the office on any given day.
- We were open to hiring remote teammates in non-engineering roles (and we soon hired some).
- Existing SF teammates could choose to move to a different location in the world if they wanted (and a few did).
- We stopped flying remote candidates to our SF office for interviews (many of them needed to be done via video chat anyway).

We did this because:

- It reflected the reality of the team and wasn't a big change.
- It was a huge benefit to our teammates to allow them to live anywhere they wanted and not have to commute. This is especially valuable for teammates with families.
- It gave teammates and candidates even more confidence that we were committed to maintaining good distributed team culture.

## All-remote

> Near the end of 2019 we decided to become all-remote and get rid of our SF office entirely. As our team grew, we didn't want to create a growing SF office subculture at the company. We wanted to treat everyone equally and didn’t want our distributed teammates to feel second class. We knew that continuing to have an office as the team grew would be an increased risk to our remote-first culture, so we decided to go all-in.

Reviewer 2: ```suggestion
Near the end of 2019, we decided to become an all-remote company and get rid of our SF office entirely. As our team grew, we didn't want to create a growing SF office subculture at the company. We wanted to treat everyone equally and didn’t want our distributed teammates to feel second class. We knew that continuing to have an office as the team grew would be an increased risk to our remote-first culture, so we decided to go all-in.
```

- Create [explicit travel budgets](https://about.sourcegraph.com/handbook/people-ops/travel) so teammates can meet up regularly (we supported travel before this transition but the budgets weren't well defined).
- Update our [employee handbook](https://about.sourcegraph.com/handbook) to codify all of the [communication tactics](https://about.sourcegraph.com/handbook/communication) that we had learned as a remote-first team.
- List, show, and negotiate a sublease for our existing office space, and get a new minimal office address for mail and state and federal paperwork (we chose to get a desk at a coworking space near our old office).
- Pack up all of our stuff (office supplies, company swag, banners for events, furniture, art and wall accessories, spare electronic equipment and cables, food, and everything in between), and decide whether we should try to store it (and if so, where), sell it, or distribute it to team members.
- Update all of our financial and legal accounts, update contracts, and more.

## Tradeoffs

> We love all-remote, but it doesn't work for everyone (just like working on an open-office plan doesn't work for everyone). Working remotely gives you extra freedom and flexibility, but also requires extra discipline and personal responsibility. Certain things that happen organically in an office environment (e.g., informal conversations at lunch), need to be explicitly planned (e.g., by scheduling small "watercooler" meetings with no agenda, or scheduling 1-1s between random teammates using a tool like donut.com).

Author: ```suggestion
We love all-remote, but it doesn't work for everyone (just like working in an open-office floorplan doesn't work for everyone). Working remotely gives you extra freedom and flexibility, but also requires extra discipline and personal responsibility. Certain things that happen organically in an office environment (e.g., informal conversations at lunch), need to be explicitly planned (e.g., by scheduling small "watercooler" meetings with no agenda, or scheduling 1-1s between random teammates using a tool like donut.com).
```

We are committed to supporting all our teammates to help them be successful. Check out our [tips for working remotely](https://about.sourcegraph.com/company/remote/tips).

> If you have any questions about all-remote teams, join us during [our Developer Meetup on 2020-04-07](https://info.sourcegraph.com/online-meetup-april-7) and I will be happy to answer them.

Author: ```suggestion
If you have any questions about all-remote teams, join us during [our Developer Meetup on Tuesday, April 7 2020 at 10:00 AM PDT](https://info.sourcegraph.com/online-meetup-april-7), and I will be happy to answer them.
```
